client/src/modules/list/components/Items/ListItemDescription/ListItemDescription.jsx

```diff
@@ -0,0 +1,285 @@
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import _trim from 'lodash/trim';
+import _isEqual from 'lodash/isEqual';
+import Linkify from 'react-linkify';
+import Textarea from 'react-textarea-autosize';
+import classNames from 'classnames';
+
+import Preloader, {
+  PreloaderSize,
+  PreloaderTheme
+} from 'common/components/Preloader';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+import { updateListItem } from '../model/actions';
+import { KeyCodes } from 'common/constants/enums';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+
+class ListItemDescription extends PureComponent {
+  pendingPromise = null;
+
+  constructor(props) {
+    super(props);
+
+    const { description } = this.props;
+    const trimmedDescription = description.trim();
+
+    this.state = {
+      isDescriptionUpdated: false,
+      descriptionInputValue: trimmedDescription,
+      isDescriptionEdited: false,
+      isFocused: false,
+      pending: false
+    };
+
+    this.descriptionTextarea = React.createRef();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    this.checkIfDescriptionUpdated();
+    const { isDescriptionEdited: previousIsDescriptionEdited } = prevState;
+    const { isDescriptionEdited } = this.state;
+
+    if (!previousIsDescriptionEdited && isDescriptionEdited) {
+      this.descriptionTextarea.current.focus();
+      this.handleFocus();
+    }
+
+    if (previousIsDescriptionEdited && !isDescriptionEdited) {
+      this.handleBlur();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  handleKeyPress = event => {
+    const { code } = event;
+    const { isFocused } = this.state;
+
+    if (isFocused && code === KeyCodes.ESCAPE) {
+      this.handleBlur();
+    }
+  };
+
+  handleFocus = () => {
+    this.setState({ isFocused: true });
+    document.addEventListener('keydown', this.handleKeyPress);
+  };
+
+  handleBlur = () => {
+    this.setState({ isFocused: false });
+    document.removeEventListener('keydown', this.handleKeyPress);
+    this.handleDescriptionUpdate();
+  };
+
+  handleStartEditing = () => {
+    this.setState({ isDescriptionEdited: true });
+  };
+
+  handleStopEditing = () => {
+    this.setState({ isDescriptionEdited: false });
+  };
+
+  handleDescriptionVisibility = () =>
+    this.setState(({ isDescriptionVisible }) => ({
+      isDescriptionVisible: !isDescriptionVisible
+    }));
+
+  handleDescriptionUpdate = () => {
+    const { descriptionInputValue: description } = this.state;
+    const {
+      description: previousDescription,
+      disabled,
+      itemId,
+      match: {
+        params: { id: listId }
+      },
+      name,
+      updateListItem
+    } = this.props;
+
+    if (disabled) {
+      return;
+    }
+
+    const isDescriptionUpdated = !_isEqual(
```

Comment:
why are you checking here if the description is updated if you have created a method for this?

this.checkIfDescriptionUpdated ?
You can set state in this method, and return a boolean value at the end and use it here